step-01/js/main.js

```diff
@@ -1,26 +1,73 @@
 'use strict';
 
-navigator.getUserMedia = navigator.getUserMedia ||
-    navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+// Define navigator.mediaDevices.getUserMedia() for older devices.
 
-var constraints = {
+// Old browsers might not implement mediaDevices, so we set an empty object.
+if (navigator.mediaDevices === undefined) navigator.mediaDevices = {};
+
+// Add mediaDevices.getUserMedia method if it's missing.
+if (navigator.mediaDevices.getUserMedia === undefined) {
+
+  /**
+   * Gets a stream of media content containing tracks of video or audio.
+   * Produces a Promise which resolves into a MediaStream object.
+   * @param {!MediaStreamConstraints} constraints to apply to the media stream.
+   * @return {!Promise} Promise which resolves to a MediaStream.
+   */
+  navigator.mediaDevices.getUserMedia = (constraints) => {
+    // Leverage legacy getUserMedia, if it is defined.
+    const getUserMedia = navigator.getUserMedia ||
+        navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+
+    // If legacy method is also not defined, resolve Promise as an error.
+    if (!getUserMedia) {
+      return Promise.reject(
+        new Error('getUserMedia is not implemented in this browser'));
+    }
+
+    // Otherwise, wrap the call to the navigator.getUserMedia with a Promise.
+    return new Promise(function(resolve, reject) {
```

Comment:
I'd prefer to use the simpler then/catch syntax like [here](https://simpl.info/getusermedia/js/main.js).

Reply:
Applied.